### PR TITLE
[AppConfiguration] Add support for ```disable_local_auth```

### PR DESCRIPTION
**Pull request by shenmuxiaosen**
**Description**
Every request to an Azure App Configuration resource must be authenticated. By default, requests can be authenticated with either Azure Active Directory (Azure AD) credentials, or by using an access key (connection string). Of these two types of authentication schemes, Azure AD provides superior security and ease of use over access keys, and is recommended by Microsoft. To require clients to use Azure AD to authenticate requests, you can disable the usage of access keys for an Azure App Configuration resource.

Once access key authentication for an Azure App Configuration resource disabled, any existing access keys for that resource are deleted. Any subsequent requests to the resource using the previously existing access keys will be rejected. Only requests that are authenticated using Azure AD will succeed.

In this pr, we will support this switch by introducing a new optional command argument ```disable_local_auth``` when creating or updating the resource. 

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by shenmuxiaosen** [comment]
@avanigupta Can you help to take look at this pr?

**Comment by shenmuxiaosen** [comment]
@Juliehzl @zhoxing-ms ```disable_local_auth``` is a JEDI requirement that most services will support. Several teams have finished the management plan sever side but I can't find any precedent in CLI. Let me know if there is a pattern that we need to follow.

**Comment by yonzhan** [comment]
AppConfiguration

File: src/azure-cli/azure/cli/command_modules/appconfig/_params.py [hunk 1]
```diff
@@ -85,11 +85,13 @@ def load_arguments(self, _):
                    help='Space-separated list of managed identities to be assigned. Use "[system]" to refer to system-assigned managed identity or a resource ID to refer to user-assigned managed identity. If this argument is provided without any value, system-assigned managed identity will be assigned by default. If this argument is not provided, no managed identities will be assigned to this App Configuration store.')
         c.argument('enable_public_network', options_list=['--enable-public-network', '-e'], arg_type=get_three_state_flag(), is_preview=True,
                    help='When true, requests coming from public networks have permission to access this store while private endpoint is enabled. When false, only requests made through Private Links can reach this store.')
+        c.argument('disable_local_auth', arg_type=get_three_state_flag(), is_preview=True, help='Disables all authentication methods other than AAD authentication.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appconfig/_params.py:88]: Please use the first person voice in help information
```suggestion
        c.argument('disable_local_auth', arg_type=get_three_state_flag(), is_preview=True, help='Disable all authentication methods other than AAD authentication.')
```

File: src/azure-cli/azure/cli/command_modules/appconfig/_params.py [hunk 2]
```diff
@@ -85,11 +85,13 @@ def load_arguments(self, _):
...
 
     with self.argument_context('appconfig update') as c:
         c.argument('tags', arg_type=tags_type)
         c.argument('enable_public_network', options_list=['--enable-public-network', '-e'], arg_type=get_three_state_flag(), is_preview=True,
                    help='When true, requests coming from public networks have permission to access this store while private endpoint is enabled. When false, only requests made through Private Links can reach this store.')
+        c.argument('disable_local_auth', arg_type=get_three_state_flag(), is_preview=True, help='Disables all authentication methods other than AAD authentication.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appconfig/_params.py:94]:
```suggestion
        c.argument('disable_local_auth', arg_type=get_three_state_flag(), is_preview=True, help='Disable all authentication methods other than AAD authentication.')
```

**Comment by zhoxing-ms** [comment]
@shenmuxiaosen At present, we use multi-API control in Azure Stack to manage the feature supported by JEDI.
doc link: [multi-api-aware-modules](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#multi-api-aware-modules
)

May I ask if this feature is only supported in the profile of JEDI or in all the profiles (include public cloud)?
Besides, are there any other requirements for JEDI?

File: src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_commands.py [hunk 3]
```diff
@@ -102,6 +102,16 @@ def test_azconfig_mgmt(self, resource_group, location):
                          self.check('sku.name', sku),
                          self.check('encryption.keyVaultProperties.keyIdentifier', keyvault_uri.strip('/') + "/keys/{}/".format(encryption_key))])
 
+        self.cmd('appconfig update -n {config_store_name} -g {rg} --disable-local-auth',
```
Comment (avanigupta) [inline src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_commands.py:105]: Can you also add a few more test cases like:
- enable local auth after disabling it
- disable local auth during store creation
- Check error message if someone tries to use HMAC auth after disabling local auth
- Check the default value of `disableLocalAuth` if someone doesn't explicitly disable local auth (is it null or false?)

File: src/azure-cli/setup.py [hunk 4]
```diff
@@ -64,7 +64,7 @@
     'azure-loganalytics~=0.1.0',
     'azure-mgmt-advisor==9.0.0',
     'azure-mgmt-apimanagement~=0.2.0',
-    'azure-mgmt-appconfiguration~=1.0.1',
+    'azure-mgmt-appconfiguration~=2.0.0',
```
Comment (avanigupta) [inline src/azure-cli/setup.py:67]: Any breaking changes that might be introduced in the 2.0.0 release?

File: src/azure-cli/azure/cli/command_modules/appconfig/custom.py [hunk 5]
```diff
@@ -34,7 +34,8 @@ def create_configstore(client,
                        location,
                        sku="Standard",
                        assign_identity=None,
-                       enable_public_network=None):
+                       enable_public_network=None,
+                       disable_local_auth=None):
```
Comment (avanigupta) [inline src/azure-cli/azure/cli/command_modules/appconfig/custom.py:38]: Could you also add some sample commands in the help file?

File: src/azure-cli/setup.py (hunk 4 above)
Comment (avanigupta) [reply src/azure-cli/setup.py:67]: Also, you'll have to add this new package to the `requirements.xxx` files too

**Comment by shenmuxiaosen** [comment]
> @shenmuxiaosen At present, we use multi-API control in Azure Stack to manage the feature supported by JEDI.
> doc link: [multi-api-aware-modules](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#multi-api-aware-modules)
> 
> May I ask if this feature is only supported in the profile of JEDI or in all the profiles (include public cloud)?
> Besides, are there any other requirements for JEDI?

This is supported in all clouds. It is a JEDI work item that each service should be able to limit requests with only AAD auth. Is there any service supporting this option in CLI? Or maybe they design in a different command.

File: src/azure-cli/azure/cli/command_modules/appconfig/_params.py (hunk 2 above)
Comment (shenmuxiaosen) [reply src/azure-cli/azure/cli/command_modules/appconfig/_params.py]: Fixed.

File: src/azure-cli/azure/cli/command_modules/appconfig/_params.py (hunk 1 above)
Comment (shenmuxiaosen) [reply src/azure-cli/azure/cli/command_modules/appconfig/_params.py]: Fixed.

File: src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_commands.py (hunk 3 above)
Comment (shenmuxiaosen) [reply src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/test_appconfig_commands.py]: Tests added

File: src/azure-cli/setup.py (hunk 4 above)
Comment (shenmuxiaosen) [reply src/azure-cli/setup.py:67]: No breaking change in this preview version. From CLI's perspective, PR just introduces some new properties.

File: src/azure-cli/azure/cli/command_modules/appconfig/custom.py (hunk 5 above)
Comment (shenmuxiaosen) [reply src/azure-cli/azure/cli/command_modules/appconfig/custom.py:38]: Sample added.

**Comment by zhoxing-ms** [comment]
> This is supported in all clouds. It is a JEDI work item that each service should be able to limit requests with only AAD auth. Is there any service supporting this option in CLI? Or maybe they design in a different command.

Sorry, I don't know much about AAD auth. @jiasli Could you help to look at the question about AAD auth?

**Review by jiasli — CHANGES_REQUESTED** [review]
I have zero knowledge about AAD auth this PR is talking about. Please add description to this PR with additional information.

**Review by avanigupta — APPROVED** [review]

**Comment by shenmuxiaosen** [comment]
> I have zero knowledge about AAD auth this PR is talking about. Please add description to this PR with additional information.

Description added.

**Comment by jiasli** [comment]
Consider suppressing Credential Scan if the credential must be recorded in the test:

https://github.com/Azure/azure-cli/blob/59aa8c23020b500e4a4d68938d59a59ff9c7e5ee/scripts/ci/credscan/CredScanSuppressions.json

**Comment by zhoxing-ms** [comment]
@shenmuxiaosen Please solve these comments and conflicts in time today, otherwise this PR will miss the release of this sprint and can only be released in the next sprint (about 08-06)

**Review by jiasli — APPROVED** [review]

**Pull request merged**